pycbc/workflow/pegasus_workflow.py

```diff
@@ -561,9 +561,12 @@ def root_path(v):
 
             # Set our needed file as output so that it gets staged upwards
             # to a workflow that contains the job which needs it.
-            for wf in input_root[:input_root.index(common)]:
-                if inp not in wf._as_job.get_outputs():
-                    wf._as_job.add_outputs(inp, stage_out=False)
+            for idx in range(input_root.index(common)):
+                child_wflow = input_root[idx]
+                parent_wflow = input_root[idx+1]
+                if inp not in child_wflow._as_job.get_outputs():
+                    child_wflow._as_job.add_outputs(inp, stage_out=True)
+                    parent_wflow._outputs += [inp]
```

Comment:
Ok, so the one complication here is that this line will fail if the file isn't meant to be staged out final output site i.e. if the file is purely intermediate then this won't work. That was in practice the case before Pegasus5 as well, just wanted to note that ideally one could use *any* file and it would do a sensible thing, not just one that is guaranteed to have a final output path. 

I'll merge as is though if that's OK, it's better than we have now and will fail on workflow generation in the case I described so at least we'll be able to identify what happened.